project/make/validate-spaces

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+#Ignoring files from vendor, ending with .md and .go"
+ignore_files='^vendor/|\.md$|\.go$'
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only | grep -v "$ignore_files" || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+    if [ "$(git show "$VALIDATE_HEAD:$f" | grep '[[:space:]]$')" ]; then
+        badFiles+=( "$f" )
+    fi
+done
+
+if [ ${#badFiles[@]} -eq 0 ]; then
+	echo 'Congratulations!  All text files are properly formatted.'
+else
+	{
+		echo "These files have trailing whitespaces:"
+		for f in "${badFiles[@]}"; do
+			echo " - $f"
+		done
+		echo
+		echo 'Please reformat the above files using, for example:'
+		echo '"ex -sc "'"%s/[[:space:]]*$//g|x"'" file" and commit the result.'
```

Comment:
I don't like the -i flag. Don't remember the exact reason but it is on the sed's man page a warning about that.

> On 10 Feb 2015, at 21:24, Tianon Gravi <notifications@github.com> wrote:
> 
> In project/make/validate-spaces:
> 
> > +    if [ "$(git show "$VALIDATE_HEAD:$f" | grep '[[:space:]]$')" ]; then
> > +        badFiles+=( "$f" )
> > +    fi
> > +done
> > +
> > +if [ ${#badFiles[@]} -eq 0 ]; then
> > +	echo 'Congratulations!  All text files are properly formatted.'
> > +else
> > +	{
> > +		echo "These files have trailing whitespaces:"
> > +		for f in "${badFiles[@]}"; do
> > +			echo " - $f"
> > +		done
> > +		echo
> > +		echo 'Please reformat the above files using, for example:'
> > +		echo '"ex -sc "'"%s/[[:space:]]*$//g|x"'" file" and commit the result.'
> IMO, ex feels like overkill here since this is something sed is really good at (and people are used to seeing/using sed).
> 
> Something simple like sed -ri 's/[[:space:]]+$//g' file should be enough to get the job done.
> 
> —
> Reply to this email directly or view it on GitHub.
>